hi-ml-cpath/src/health_cpath/scripts/generate_checkpoint_url.py

```diff
@@ -19,24 +20,23 @@ def get_checkpoint_url_from_aml_run(
 
     :param run_id: The run ID of the checkpoint.
     :param checkpoint_filename: The filename of the checkpoint.
+    :param account_key: The Azure Storage account key to use for the SAS token.
     :param expiry_days: The number of days the SAS URL is valid for, defaults to 30.
     :param workspace_config_path: The path to the workspace config file, defaults to None.
     :param aml_workspace: The Azure ML workspace to use, defaults to the default workspace.
     :param sas_token: The SAS token to use, defaults to None.
     :return: The SAS URL for the checkpoint.
     """
     workspace = get_workspace(aml_workspace=aml_workspace, workspace_config_path=workspace_config_path)
-    datastore = workspace.get_default_datastore()
-    account_name = datastore.account_name
+    account_name = workspace.get_details()['storageAccount'].split('/')[-1]
```

Comment:
It would be helpful here to print out the name of the storage account. Chances are high that the account key does not match. something like "Workspace {workspace.name} stores its run results in storage account {account_name}."

Reply:
done